src/gas/GasFeeController.ts

```diff
@@ -1,10 +1,9 @@
 import type { Patch } from 'immer';
```

Comment:
This seems like a lot, so I'll try to summarize:

GasFeeController was designed for polling for gas fee estimates. These changes extend the polling logic such that it is now also possible to poll for network congestion. We achieve this by removing the concept of poll tokens and replacing this with known pollable items ("gasFeeEstimates" and "networkCongestion") as well as a poll queue (a mapping of pollable items to functions).

One could argue that these changes do not belong in this controller, and one *may* be correct. However, I tried to put the bits about network congestion / fee history in a separate controller, and the problem is that if we want to poll for two things at the same time, that makes that really hard to do. It seems that we might could benefit from a PollingController, but I really wasn't sure what the best approach here, so it seemed easier to extend this controller. Let me know if you have any alternative opinions though!

Reply:
Haven't read through the changes here yet but, in theory, including polling for 'networkCongestion' in this controller makes sense to me (especially given, as you point out, we intend to poll the same RPC method as an estimate fee estimate fallback.

Reply:
Resolving this conversation since I ended up coming up with a simpler solution.